cmake/developer_package/compile_flags/os_flags.cmake

```diff
@@ -262,34 +300,53 @@ if(ENABLE_COVERAGE)
     set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
 endif()
 
-if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    ie_add_compiler_flags(-fsigned-char)
-endif()
-
 # Honor visibility properties for all target types
 set(CMAKE_POLICY_DEFAULT_CMP0063 NEW)
 set(CMAKE_CXX_VISIBILITY_PRESET hidden)
 set(CMAKE_C_VISIBILITY_PRESET hidden)
 set(CMAKE_VISIBILITY_INLINES_HIDDEN ON)
 
-function(ie_python_minimal_api target)
-    # pybind11 uses a lot of API which is not a part of minimal python API subset
-    # Ref 1: https://docs.python.org/3.11/c-api/stable.html
-    # Ref 2: https://github.com/pybind/pybind11/issues/1755
-    # target_compile_definitions(${target} PRIVATE Py_LIMITED_API=0x03090000)
-    # if(NOT CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
-    #     target_compile_options(${target} PRIVATE "-Wno-unused-variable")
-    # endif()
-endfunction()
+if(CMAKE_CL_64)
+    # Default char Type Is unsigned
+    ie_add_compiler_flags(/J)
+else()
+    ie_add_compiler_flags(-fsigned-char)
+endif()
 
 if(WIN32)
-    ie_add_compiler_flags(-D_CRT_SECURE_NO_WARNINGS -D_SCL_SECURE_NO_WARNINGS)
-    ie_add_compiler_flags(/EHsc) # no asynchronous structured exception handling
-    ie_add_compiler_flags(/Gy) # remove unreferenced functions: function level linking
+    #
+    # Common options / warnings enabled
+    #
+
+    ie_add_compiler_flags(/D_CRT_SECURE_NO_WARNINGS /D_SCL_SECURE_NO_WARNINGS)
+     # no asynchronous structured exception handling
+    ie_add_compiler_flags(/EHsc)
+    # Allows the compiler to package individual functions in the form of packaged functions (COMDATs).
+    ie_add_compiler_flags(/Gy)
+    # This option helps ensure the fewest possible hard-to-find code defects. Similar to -Wall on GNU / Clang
+    ie_add_compiler_flags(/W3)
+
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+        # Increase Number of Sections in .Obj file
+        ie_add_compiler_flags(/bigobj)
```

Comment:
Why do we need these options?

Reply:
`Increase Number of Sections in .Obj file`